### PR TITLE
Create API Endpoint: GET/POST List of Groups + Detail View of User Groups

### LINKED ISSUE #372: Create API Endpoint: GET/POST List of Groups + Detail View of Groups
**Goal**

Set up an endpoint that retrieves information about a list of ChiGame Groups from API, serialized in JSON. This will be done at url `api/groups/`, and can be accessed by ChiGame users as well as third-party websites. The games GET request would include a group name, group members, the user it was created by, and the date created. 

ENDPOINT 

`/api/groups`: Supports GET and POST requests in order to get a list of groups (and the corresponding information) as well as be able to create new groups using the API

We will know this is completed when the functionality above is working.

### PR DESCRIPTION
**Pull request by abrahmasandra**
Created an endpoint that retrieves information about a list of ChiGame Groups from API, serialized in JSON. This will be done at url `api/groups/`, and can be accessed by ChiGame users as well as third-party websites. The games GET request would include a group name, group members, the user it was created by, and the date created. Also created endpoint that returns the detail view of groups (`/api/groups/<group_id>`)

ENDPOINT 

`/api/groups`: Supports GET and POST requests in order to get a list of groups (and the corresponding information) as well as be able to create new groups using the API

`/api/groups/<int: group_id>`: Supports GET/PATCH/DELETE requests in order to get information or change information about a specific group

### CODE REVIEW

File: src/chigame/api/urls.py
Comment (cuyakonwu) [inline src/chigame/api/urls.py:1]: Same as another comment I made, might be worth making it a variable to help with organization

**Review by cuyakonwu — APPROVED** [review]
Great work here!

**Review by majorsylvie — APPROVED** [review]
Can successfully view list and individual groups.
Can POST on the list view, as well as PUT, and DELETE on the detail view.

Works as expected. Thank you for your work.

**Comment by abrahmasandra** [comment]
Closing Comment: 

Created two views (the list view and detail view) for the `Group` model. Specifically, created the endpoint `/api/groups` for list view, supporting `GET`/`POST` HTTP requests. Created the endpoint `/api/groups/<group_id>` to get the detail view of a particular group from the given `group_id` (endpoint supports `GET`/`PATCH`/`DELETE` requests)

**Comment by majorsylvie** [comment]
Issue Score: Excellent

**Pull request merged**